### PR TITLE
Call Ansible for just one domain cert generation

### PR DESCRIPTION
**Pull request by melvinsoft**
This PR calls the ansible playbooks that generation the custom domain certificate, sending the new domain as a parameter, so we can generate the cert for a single domain. This PR works together with https://github.com/appsembler/configuration/pull/161 and https://github.com/appsembler/roles/pull/46/files.

The workflow switch happens when the bool `letsencrypt_execute_for_single_domain` is `true` this is set in https://github.com/appsembler/roles/pull/46/files, in the server vars defined for certbot.

### CODE REVIEW

**Review by thraxil — CHANGES_REQUESTED** [review]
I think if we're getting to where `domain` is getting actually passed into a shell command, we need to be a little more explicit about making sure that it is safe. There's a validator earlier that checks that it is a validly formatted domain, which rules out a lot of potential problems but 1) I don't know if that check is written with security in mind and 2) we might refactor at some point and move that validation elsewhere and end up with this unprotected, so I think it's worth being clear.

At a minimum, we should strip out anything that definitely can't be in a domain.

Ideally, it might be worth switching `settings.ANSIBLE_CMD` to be a list rather than string and figure out if we can get away with setting `shell=False`.

Basically, when `settings.ANSIBLE_CMD` was the only thing being passed to `subprocess.Popen()`, the risk was minimal. Now that user supplied data is going through, it's time to go full paranoid.

**Comment by melvinsoft** [comment]
@thraxil Agreed, I'm gonna work on this tomorrow! Thanks!

**Comment by bezidejni** [comment]
I agree with Anders, there's too much potential for abuse right now.

**Comment by melvinsoft** [comment]
@thraxil @bezidejni As we discussed in slack, this changes are necessary, but we're in a rush and the feature is still behind a feature flag.

Do you think we can merge this with the current level of checks, and continue working before open it to the public?

I've created an Epic to address this soon: https://trello.com/c/xXtgYDLC/171-custom-domains-stability-improvements-part-2

**Comment by thraxil** [comment]
@melvinsoft I added commit b05cf48a1f8e to do a quick stripping of any characters that aren't allowed in the domain and could thus be harmful. It's nothing fancy, but it makes it clear that caution needs to be exercised when passing a variable out to a shell.

**Comment by bezidejni** [comment]
I'm fine with merging it after Anders added a check, in the future we might want to use the [validators](https://validators.readthedocs.io/en/latest/) library, it's really useful.

**Review by bezidejni — APPROVED** [review]

**Review by thraxil — APPROVED** [review]

**Comment by melvinsoft** [comment]
@thraxil Thanks for pushing those commits and @bezidejni Thanks for your review !
